support-frontend/app/utils/CheckoutValidationRules.scala

```diff
@@ -6,84 +6,116 @@ import com.gu.support.catalog.{Collection, Domestic, FulfilmentOptions, HomeDeli
+    case existingDetails: ExistingPaymentFields => existingDetails.billingAccountId.nonEmpty.otherwise("existing billing account id missing")
   }
 
 }
 
 object AddressAndCurrencyValidationRules {
 
-  def deliveredToUkAndPaidInGbp(countryFromRequest: Country, currencyFromRequest: Currency): Boolean =
-    countryFromRequest == Country.UK && currencyFromRequest == GBP
+  def deliveredToUkAndPaidInGbp(countryFromRequest: Country, currencyFromRequest: Currency): Result =
+    (countryFromRequest == Country.UK).otherwise(s"should be delivered to UK, was $countryFromRequest") and
+      (currencyFromRequest == GBP).otherwise(s"should be paid for in GBP, was $currencyFromRequest")
 
-  def hasAddressLine1AndCity(address: Address): Boolean = {
-    address.lineOne.isDefined && address.city.isDefined
+  def hasAddressLine1AndCity(address: Address): Result = {
+    address.lineOne.isDefined.otherwise("must have address line 1") and
+      address.city.isDefined.otherwise("must have a city")
   }
 
 
-  def hasStateIfRequired(countryFromRequest: Country, stateFromRequest: Option[String]): Boolean =
+  def hasStateIfRequired(countryFromRequest: Country, stateFromRequest: Option[String]): Result =
     if (countryFromRequest == Country.US || countryFromRequest == Country.Canada || countryFromRequest == Country.Australia) {
-      stateFromRequest.isDefined
-    } else true
+      stateFromRequest.isDefined.otherwise(s"state is required for $countryFromRequest")
+    } else Valid
 
 
-  def hasPostcodeIfRequired(countryFromRequest: Country, postcodeFromRequest: Option[String]): Boolean =
+  def hasPostcodeIfRequired(countryFromRequest: Country, postcodeFromRequest: Option[String]): Result =
     if (
       countryFromRequest == Country.UK ||
         countryFromRequest == Country.US ||
         countryFromRequest == Country.Canada ||
         countryFromRequest == Country.Australia
     ) {
-      postcodeFromRequest.isDefined
-    } else true
+      postcodeFromRequest.isDefined.otherwise(s"postcode is required for $countryFromRequest")
+    } else Valid
 
 
-  def currencyIsSupportedForCountry(countryFromRequest: Country, currencyFromRequest: Currency): Boolean = {
+  def currencyIsSupportedForCountry(countryFromRequest: Country, currencyFromRequest: Currency): Result = {
     val countryGroupsForCurrency = CountryGroup.allGroups.filter(_.currency == currencyFromRequest)
-    countryGroupsForCurrency.flatMap(_.countries).contains(countryFromRequest)
+    val isValid = countryGroupsForCurrency.flatMap(_.countries).contains(countryFromRequest)
+
+    val validCurrenciesForCountry = CountryGroup.allGroups.filter(_.countries.contains(countryFromRequest)).flatMap { countryGroup =>
+      countryGroup.currency :: countryGroup.additionalCurrencies
+    }
+    isValid.otherwise(s"currency $currencyFromRequest is not supported for country $countryFromRequest, can only have ${validCurrenciesForCountry}")
```

Comment:
this logic was a bit upside down given the function name. It actually finds the currency and checks the country is valid for the currency.  I have kept that logic but done the calculation the other way round for the error message, rather than refactoring the logic in this PR.  However - opinions welcome!